api/v1/tetragon/tetragon.proto

```diff
@@ -45,71 +50,128 @@ message Container {
+    // added to the inheritable set by a thread without the CAP_SETPCAP in its
+    // effective set.
     repeated CapabilitiesType permitted = 1;
+    // Effective set indicates what capabilities are active in a process. This
+    // is the set used by the kernel to perform permission checks for the
+    // thread.
     repeated CapabilitiesType effective = 2;
+    // Inheritable set indicates which capabilities will be inherited by the
+    // current process when running as a root user.
     repeated CapabilitiesType inheritable = 3;
 }
 
 message Namespace {
+    // Inode number of the namespace.
     uint32 inum = 1;
+    // Indicates if namespace belongs to host.
     bool is_host = 2;
 }
 
 message Namespaces {
+    // Hostname and NIS domain name.
     Namespace uts = 1;
+    // System V IPC, POSIX message queues.
     Namespace ipc = 2;
+    // Mount points.
     Namespace mnt = 3;
+    // Process IDs.
     Namespace pid = 4;
+    // Process IDs for children processes.
     Namespace pid_for_children = 5;
+    // Network devices, stacks, ports, etc.
     Namespace net = 6;
+    // Boot and monotonic clocks.
     Namespace time = 7;
+    // Boot and monotonic clocks for children processes.
     Namespace time_for_children = 8;
+    // Cgroup root directory.
     Namespace cgroup = 9;
+    // User and group IDs.
     Namespace user = 10;
 }
 
 message Process {
     // Exec ID uniquely identifies the process over time across all the nodes in the cluster.
     string exec_id = 1;
+    // Process identifier from host PID namespace.
     google.protobuf.UInt32Value pid = 2;
+    // User identifier associated with the process.
     google.protobuf.UInt32Value uid = 3;
+    // Current working directory of the process.
     string cwd = 4;
+    // Absolute path of the executed binary.
     string binary = 5;
+    // Arguments passed to the binary at execution.
     string arguments = 6;
+    // Information about which syscalls generated events, use of internal
+    // Tetragon buffers, errors and more.
+    // TODO: not sure how to documents all the existing flags without dumping
+    // 60 lines of text
     string flags = 7;
+    // Start time of the execution.
     google.protobuf.Timestamp start_time = 8;
+    // TODO: what is this identifier about? related to uid apparently.
     google.protobuf.UInt32Value auid = 9;
+    // Information about the the Kubernetes Pod where the event originated.
     Pod pod = 10;
+    // TODO: is this the 15 first digits of the docker ID? We should name it
+    // something else than docker since most k8s install no longer use Docker.
     string docker = 11;
+    // Exec ID of the parent process.
     string parent_exec_id = 12;
+    // TODO: is this a similar refcnt than from the BPF stats?
     uint32 refcnt = 13;
+    // Set of capabilities that define the permissions the process can execute with.
     Capabilities cap = 14;
+    // TODO: is this used?
     Namespaces ns = 15;
```

Comment:
Do you know when it's used?

Reply:
https://github.com/cilium/tetragon/blob/bc5f11a52c6bc19607dc6c15aa7fe44894e86a1b/pkg/process/process.go#L134-L136

Reply:
ah okay there is a `option.Config.EnableProcessNs`